enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/pipes/RollUpApplySlottedPipe.scala

```diff
@@ -0,0 +1,84 @@
+ */
+package org.neo4j.cypher.internal.compatibility.v3_4.runtime.slotted.pipes
+
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime._
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.commands.expressions.Expression
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.pipes.{Pipe, PipeWithSource, QueryState}
+import org.neo4j.cypher.internal.compatibility.v3_4.runtime.slotted.PrimitiveExecutionContext
+import org.neo4j.cypher.internal.v3_4.logical.plans.LogicalPlanId
+import org.neo4j.values.AnyValue
+import org.neo4j.values.storable.Values.NO_VALUE
+import org.neo4j.values.virtual.VirtualValues
+
+case class RollUpApplySlottedPipe(lhs: Pipe, rhs: Pipe,
+                                  collectionName: String,
+                                  identifierToCollect: (String, Expression),
+                                  nullableIdentifiers: Set[String],
+                                  pipelineInformation: PipelineInformation)
+                                 (val id: LogicalPlanId = LogicalPlanId.DEFAULT)
+  extends PipeWithSource(lhs) {
+
+  private val collectionSlot = pipelineInformation.get(collectionName).get
+
+  private val getValueToCollectFunction = {
+    val expression: Expression = identifierToCollect._2
+    (state: QueryState) => (ctx: ExecutionContext) => expression(ctx, state)
+  }
+
+  private val hasNullValuePredicates: Seq[(ExecutionContext) => Boolean] =
+    nullableIdentifiers.toSeq.map { elem =>
+      val elemSlot = pipelineInformation.get(elem)
+      elemSlot match {
+        case Some(LongSlot(offset, true, _, _)) => { (ctx: ExecutionContext) => ctx.getLongAt(offset) == -1 }
+        case Some(RefSlot(offset, true, _, _)) => { (ctx: ExecutionContext) => ctx.getRefAt(offset) == NO_VALUE }
+        case _ => { (ctx: ExecutionContext) => false }
+      }
+    }
+
+  private def hasNullValue(ctx: ExecutionContext): Boolean =
+    hasNullValuePredicates.exists(p => p(ctx))
+
+  private val setCollectionInRow: (ExecutionContext, AnyValue) => Unit = {
+    collectionSlot match {
+      case RefSlot(offset, _, _, _) =>
+        (ctx: ExecutionContext, value: AnyValue) => ctx.setRefAt(offset, value)
+      case _ => throw new InternalError("Expected collection to be allocated to a ref slot")
+    }
+  }
+
+  override protected def internalCreateResults(input: Iterator[ExecutionContext], state: QueryState) = {
+    input.map {
+      ctx =>
+        val outputRow = PrimitiveExecutionContext(pipelineInformation)
+        ctx.copyTo(outputRow)
+        if (hasNullValue(ctx)) {
+          setCollectionInRow(outputRow, NO_VALUE)
+        }
+        else {
+          val innerState = state.withInitialContext(ctx)
+          val innerResults: Iterator[ExecutionContext] = rhs.createResults(innerState)
+          val collection = VirtualValues.list(innerResults.map(getValueToCollectFunction(state)).toArray: _*)
+          setCollectionInRow(outputRow, collection)
```

Comment:
Instead of using a method value, I think we should calculate the offset to an int at constructor time and just `outputRow.setRefAt` here. This is in the tight loop, so minimizing virtual calls is a good idea.